depscan/lib/csaf.py

```diff
@@ -0,0 +1,1663 @@
+    elif len(pkg) == 4:
+        pkg = matched_by.split("|")[2]
+    if issue.get("fixed_location"):
+        product_status["fixed"] = [f"{pkg}:{issue.get('fixed_location')}"]
+    if issue.get("affected_location"):
+        try:
+            loc_dict = issue.get("affected_location")
+            product_status["known_affected"] = [
+                f'{loc_dict.get("package")}:{loc_dict.get("version")}'
+            ]
+        except json.JSONDecodeError:
+            logging.warning("Invalid JSON string for affected_location")
+    return pkg, product_status
+
+
+def parse_cwe(cwe):
+    fmt_cwe = None
+    new_notes = []
+
+    if not cwe or cwe in ["UNKNOWN", [], "[]"]:
+        return fmt_cwe, new_notes
+
+    cwe_ids = re.findall(r"CWE-[1-9]\d{0,5}", cwe)
+    for i in range(len(cwe_ids)):
+        cwe_name = CWE_MAP.get(cwe_ids[i], "UNABLE TO LOCATE CWE NAME")
+        if not cwe_name:
+            LOG.warning(
+                f"We couldn't locate the name of the CWE with the following "
+                f"id: {cwe_ids[i]}. Help us out by reporting the id at "
+                f"https://github.com/owasp-dep-scan/dep-scan/issues."
+            )
+        if i == 0:
+            fmt_cwe = {
+                "id": cwe_ids[i],
+                "name": cwe_name,
+            }
+        # CSAF 2.0 only allows a single CWE per vulnerability, so we add
+        # any additional CWEs to a note entry.
+        else:
+            new_notes.append(
+                {
+                    "title": f"Additional CWE: {cwe_ids[i]}",
+                    "audience": "developers",
+                    "category": "other",
+                    "text": cwe_name,
+                }
+            )
+
+    return fmt_cwe, new_notes
+
+
+def parse_cvss(res):
+    """
+    Parses the CVSS information from the given response.
+
+    Parameters:
+        res (dict): The response containing the CVSS information.
+
+    Returns:
+        dict or None: The parsed CVSS information as a dictionary, or None if
+        the CVSS vector string is empty as it is required for cvss v3.
+            The dictionary contains the following keys:
+                - baseScore (float): The base score of the CVSS.
+                - attackVector (str): The attack vector of the CVSS.
+                - privilegesRequired (str): Privileges required for the CVSS.
+                - userInteraction (str): User interaction required for the CVSS.
+                - scope (str): The scope of the CVSS.
+                - impactScore (str): The impact score of the CVSS.
+                - baseSeverity (str): The base severity of the CVSS.
+                - version (str): The version of the CVSS.
+                - vectorString (str): The vector string of the CVSS.
+            If the CVSS vector string is empty, None is returned.
+    """
+    cvss_v3 = res["cvss_v3"]
+    if not cvss_v3.get("vector_string"):
+        LOG.warning(f"Could not find vector string for {res['id']}")
```

Comment:
Do we want these to be warnings or info?

Reply:
@prabhu Thanks, I'll remove that - I think this was for my testing.